core-bundle/src/Command/MigrateCommand.php

```diff
@@ -186,6 +196,30 @@ private function executeCommand(InputInterface $input): int
         return 0;
     }
 
+    private function hasWorkToDo(): bool
+    {
+        // There are some pending migrations
+        foreach ($this->migrations->getPending() as $migration) {
```

Comment:
```suggestion
        foreach ($this->migrations->getPending() as $migration) { // phpcs:ignore
```

Does this work?

Reply:
Unfortunately, it does not. I also tried other comments, but it seems that all inline annotations are ignored.

Shall I add `core-bundle/src/Command/MigrateCommand.php` to the `tools/ecs/config/default.php` to ignore the whole file there?

Reply:
Yes, but only for the specific rule I guess: https://github.com/contao/contao/blob/12c7f37523f4f4bfedaaccb1b322ade711dab52c/tools/ecs/config/default.php#L24

Reply:
Done.

Reply:
Why not `if ($this->migrations->getPending()) { return true; }`? This would save us another iteration:

With foreach: https://3v4l.org/fIM4m
With if: https://3v4l.org/qDPLi

Reply:
Improved in [df8a88e](https://github.com/contao/contao/pull/5104/commits/df8a88e80bf389313a04333dd4d1c67d9e240da5).

Reply:
But `if ($this->migrations->getPending())` would always be true, even if no migrations are pending.
(The unit tests probalby don’t fail because they don’t return a real generator for `getPending()`)

Reply:
https://3v4l.org/m4KO5

Reply:
I see. How about using `$this->migrations->getPending()->current()` then?

https://3v4l.org/2nmVd

Reply:
PHPStan would probably complain because `iterable` can also be an array or an `IteratorAggregate` which have no `current()` method...

Reply:
After discussing this with @ausi, we concluded that adding a `hasPending()` method is probably the easiest fix. I have therefore restored @qzminski's initial version in 267b1cb076cf832d74629eee06f2a9035854c71b. Sorry for the back and forth. 🙈